icu4c/source/i18n/usearch.cpp

```diff
@@ -565,16 +565,26 @@ inline void initialize(UStringSearch *strsrch, UErrorCode *status)
 #if BOYER_MOORE
 /**
 * Check to make sure that the match length is at the end of the character by
-* using the breakiterator.
+* using the internal breakiterator, creating it if needed.
 * @param strsrch string search data
 * @param start target text start offset
 * @param end target text end offset
 */
 static
 void checkBreakBoundary(const UStringSearch *strsrch, int32_t * /*start*/,
-                               int32_t *end)
+                               int32_t *end, UErrorCode *status)
 {
 #if !UCONFIG_NO_BREAK_ITERATION
+    // Lazily create the internal break iterator.
```

Comment:
This is inside `#if BOYER_MOORE`, which is hardcoded to 0 at line 31.
Also, wherever the lazy creation ends up, would want to check for a caller-supplied break iterator, `strsrch->search->breakIter`, before creating an internal one.

Reply:
Thanks for taking a look!

Right, this is inside the `BOYER_MOORE` parts of the code that don't seem to be used at all anymore. However, I thought that it would be better to try and change all the places in the file that used the break iterator anyways...

Looking at the history in git, it seems that `usearch.cpp` hasn't actually compiled with the macro `BOYER_MOORE` set to `1` since the Collation V2 rewrite in ~2014. (It tries to use functions that were removed/changed since then). So perhaps this is a pointless change -- but I at least didn't want to make things any worse off in case somebody in the future wanted to try and get the `BOYER_MOORE` code working again.

Good point about checking for the caller-supplied break iterator first!
I'll force-push an update for this in a bit.